### ISSUE TITLE
Compile errors with SPIFFS and new warnings enabled (IDFGH-10696)

### ISSUE THREAD

**Issue by danfleck** [espressif/esp-idf#11922]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.1

### Operating System used.

macOS

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

None

### What is the expected behavior?

When enabling the SPIFFS debug checks, new compiler warnings cause the build to fail. 

This is due to the new compiler warnings and the fix is described here: https://github.com/espressif/esp-idf/issues/10379#issuecomment-1353302879

### What is the actual behavior?

Compile fails with warnings about format

..../esp/esp-idf-v5.1/components/log/include/esp_log.h:265:27: error: format '%d' expects argument of type 'int', but argument 8 has type 'u32_t' {aka 'long unsigned int'} [-Werror=format=]
  265 | #define LOG_COLOR(COLOR)  "\033[0;" COLOR "m"

### Steps to reproduce.

1. Create a new project using idf.py create-project
2. Set the SDK config defaults to:
```
# This file was generated using idf.py save-defconfig. It can be edited manually.
# Espressif IoT Development Framework (ESP-IDF)  Project Minimal Configuration
#
CONFIG_SPIFFS_DBG=y
CONFIG_SPIFFS_API_DBG=y
CONFIG_SPIFFS_GC_DBG=y
CONFIG_SPIFFS_CACHE_DBG=y
CONFIG_SPIFFS_CHECK_DBG=y
CONFIG_SPIFFS_TEST_VISUALISATION=y
```
3. idf.py build

### Build or installation Logs.

```plain
Many errors of this type

.../esp-idf-v5.1/components/spiffs/spiffs/src/spiffs_nucleus.c:2226:1: error: format '%d' expects argument of type 'int', but argument 8 has type 'u32_t' {aka 'long unsigned int'} [-Werror=format=]
 2226 |       SPIFFS_DBG("free_obj_id: COMP min:"_SPIPRIid" max:"_SPIPRIid" compact:"_SPIPRIi"\n", state.min_obj_id, state.max_obj_id, state.compaction);
      | ^     ~~~~~
.../esp-idf-v5.1/components/log/include/esp_log.h:282:59: note: in definition of macro 'LOG_FORMAT'
  282 | #define LOG_FORMAT(letter, format)  LOG_COLOR_ ## letter #letter " (%" PRIu32 ") %s: " format LOG_RESET_COLOR "\n"
      |                                                           ^~~~~~
.../esp-idf-v5.1/components/log/include/esp_log.h:432:41: note: in expansion of macro 'ESP_LOG_LEVEL'
  432 |         if ( LOG_LOCAL_LEVEL >= level ) ESP_LOG_LEVEL(level, tag, format, ##__VA_ARGS__); \
      |                                         ^~~~~~~~~~~~~
.../esp-idf-v5.1/components/log/include/esp_log.h:343:38: note: in expansion of macro 'ESP_LOG_LEVEL_LOCAL'
  343 | #define ESP_LOGD( tag, format, ... ) ESP_LOG_LEVEL_LOCAL(ESP_LOG_DEBUG,   tag, format, ##__VA_ARGS__)
      |                                      ^~~~~~~~~~~~~~~~~~~
.../esp-idf-v5.1/components/spiffs/include/spiffs_config.h:35:37: note: in expansion of macro 'ESP_LOGD'
   35 | #define SPIFFS_DBG(...)             ESP_LOGD(SPIFFS_TAG, __VA_ARGS__)
      |                                     ^~~~~~~~
.../esp-idf-v5.1/components/spiffs/spiffs/src/spiffs_nucleus.c:2226:7: note: in expansion of macro 'SPIFFS_DBG'
 2226 |       SPIFFS_DBG("free_obj_id: COMP min:"_SPIPRIid" max:"_SPIPRIid" compact:"_SPIPRIi"\n", state.min_obj_id, state.max_obj_id, state.compaction);
      |       ^~~~~~~~~~
```

**Comment by fink-at-trmc-dk** [comment]
Ran into this one as well...
Same fix...

**Comment by josesimoes** [comment]
The fix in https://github.com/espressif/esp-idf/commit/c99a036b437473a9656839174d634ca772a86e94 
`#define _SPIPRIi "%"PRIdMAX` is breaking the build to me.... From where `PRIdMAX` is comming from? 😯  I'm guessing it slipped into the commit...

**Comment by adokitkat** [comment]
> The fix in [c99a036](https://github.com/espressif/esp-idf/commit/c99a036b437473a9656839174d634ca772a86e94) `#define _SPIPRIi "%"PRIdMAX` is breaking the build to me.... From where `PRIdMAX` is comming from? 😯 I'm guessing it slipped into the commit...

Hi, could you post your idf version and output log here please?

**Comment by josesimoes** [comment]
IDF 5.1.2

```text
[build] E:/GitHub/esp-idf/components/spiffs/include/spiffs_config.h:74:20: error: invalid suffix on literal; C++11 requires a space between literal and string macro [-Werror=literal-suffix]
[build]    74 | #define _SPIPRIi   "%"PRIdMAX
[build]       |                    ^
[build] cc1plus.exe: all warnings being treated as errors
```

Adding an include to <inttypes.h> does not help.

In, instead, there is a space before `PRIdMAX` the compiler is happy:
`#define _SPIPRIi   "%" PRIdMAX`
